app/src/main/java/com/example/jason/dinner_rush/TradeActivity.java

```diff
@@ -1,13 +1,97 @@
 package com.example.jason.dinner_rush;
 
-import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+
+import java.util.Hashtable;
 
 public class TradeActivity extends AppCompatActivity {
 
+    // Assuming somehow we get this from the other play activity
+    Hashtable<String,Ingredient> ingredients = new Hashtable<String,Ingredient>();
+
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_trade);
+
+        // TODO: get ingredient's current hashtable from other activities
+        // ingredients = savedInstanceState.blahblahblah
+    }
+
+    void receiveMessage() {
+        // TODO: Listener for sent ingredients and acks
+        // If it's an ingredient, calls receiveIngredient() when it gets one.
+        // If it's an ACK, decrease that ingredient's amount by 1 in AckSentIngredient();
+    }
+
+    void sendIngredient(String name) {
+        // Send string (example: send "carrot")
+
+        // Current ingredient
+        Ingredient cur_ig = ingredients.get(name);
+
+        if (cur_ig == null || cur_ig.quantity <= 0) {

```

Comment:
One or the other. Either populate the HashTable with all possible ingredients, or remove ingredients from the table when they run out.